lib/minifiers/bun.ts

```diff
@@ -0,0 +1,19 @@
+import { spawn } from 'child_process';
+import type { Stream } from 'stream';
+import { minifier } from '../types';
+
+const stream2buffer = (
+	stream: Stream,
+) => new Promise<Buffer>((resolve, reject) => {
+	const bufferChunks = new Array<any>();
+
+	stream.on('error', reject);
+	stream.on('data', chunk => bufferChunks.push(chunk));
+	stream.on('end', () => resolve(Buffer.concat(bufferChunks)));
+});
+
+export default minifier(async ({ filePath }) => {
+	const minify = spawn('bun', ['build', '--no-bundle', '--minify', filePath]);
+	const minified = await stream2buffer(minify.stdout);
+	return minified.toString();
```

Comment:
This also benchmarks the time it takes for spawn to read the stream and convert the stream to text. The text decoding step adds up for large amounts of text. I suggest converting the output to a string separately and omitting that, for each benchmark (keep in mind that esbuild's API has a lazy getter to convert it to text so you would want to do that there too)

Reply:
Good tip! Thanks, will do that for every minifier in a follow up PR.

Reply:
I looked into this and it doesn't look like esbuild has a lazy getter.

```ts
const minified = await esbuild.transform(code, {
	minify: true,
	sourcemap: false,
	legalComments: 'none',
	treeShaking: true,
});

console.log(Object.getOwnPropertyDescriptors(minified));

/*
{
  warnings: {
    value: [],
    writable: true,
    enumerable: true,
    configurable: true
  },
  code: {
    value: '...',
    writable: true,
    enumerable: true,
    configurable: true
  },
  map: {
    value: '',
    writable: true,
    enumerable: true,
    configurable: true
  },
  mangleCache: {
    value: undefined,
    writable: true,
    enumerable: true,
    configurable: true
  },
  legalComments: {
    value: undefined,
    writable: true,
    enumerable: true,
    configurable: true
  }
}
*/
```

Reply:
ah maybe that's in bundle(). The one that has a .text method

Reply:
Yeah looks like the bundler:
https://github.com/evanw/esbuild/blob/b42b64540b74a01d02a54d4e4814b9a76a5cf746/lib/shared/common.ts#L1758

Reply:
Upon deeper consideration, I'm thinking it's fairer to expect a string output from each minifier in the benchmark.

If we allow minifiers to return a buffer, it would give binary minifiers an unfair advantage over those that work within Node.js (e.g., Uglify), while also penalizing binary minifiers that already does the conversion (e.g., esbuild).

By requiring a string output from all minifiers, we ensure a level playing field with consistent limitations for each one.

Reply:
Reasonable 

👍